### PR TITLE
Fix indentation from string or comment

### LINKED ISSUE #24: indentation problem
https://github.com/d12frosted/fish-mode/issues/2

### PR DESCRIPTION
**Pull request by alphapapa**
Fixes #24.

### CODE REVIEW

File: fish-mode.el [hunk 1]
```diff
@@ -387,7 +389,7 @@ For example, (fold F X '(1 2 3)) computes (F (F (F X 1) 2) 3)."
 
 (defun fish/line-contains-open-switch-term? ()
   "Returns t if line contains switch term, nil otherwise."
-  (> (fish/count-of-tokens-in-string (rx symbol-start "switch" symbol-end) nil (fish/current-line))
+  (> (fish/count-of-tokens-on-current-line (rx symbol-start "switch" symbol-end) nil)
```
Comment (wwwjfy) [inline fish-mode.el]: `fish/count-of-tokens-on-current-line` should be `fish/count-tokens-on-current-line`

File: fish-mode.el [hunk 2]
```diff
@@ -335,18 +338,18 @@ For example, (fold F X '(1 2 3)) computes (F (F (F X 1) 2) 3)."
...
+POSITIVE-RE and NEGATIVE-RE are regular expressions."
+  (save-excursion
+    (cl-loop initially (goto-char (line-beginning-position))
+             while (re-search-forward positive-re (line-end-position) t)
+             count (not (or (when negative-re
+                              (looking-at-p negative-re))
+                            ;; String
+                            (nth 3 (syntax-ppss))
+                            ;; Comment
+                            (nth 4 (syntax-ppss)))))))
```
Comment (wwwjfy) [inline fish-mode.el]: I don't know if saving `(syntax-ppss)` to a variable can improve performance

File: fish-mode.el [hunk 3]
```diff
@@ -335,18 +338,18 @@ For example, (fold F X '(1 2 3)) computes (F (F (F X 1) 2) 3)."
...
-        (setq pos nil)))
-    count))
+(defun fish/count-tokens-on-current-line (positive-re negative-re)
+  "Return count of matches for POSITIVE-RE that do not also match NEGATIVE-RE on current line.
+POSITIVE-RE and NEGATIVE-RE are regular expressions."
+  (save-excursion
+    (cl-loop initially (goto-char (line-beginning-position))
+             while (re-search-forward positive-re (line-end-position) t)
+             count (not (or (when negative-re
+                              (looking-at-p negative-re))
```
Comment (wwwjfy) [inline fish-mode.el]: This seems problematic.
e.g.

```fish
if false; echo 2; else if true; echo 2; end
```

Comment (alphapapa) [inline fish-mode.el]: Thanks, good catch, fixing...

File: fish-mode.el (hunk 2 above)
Comment (alphapapa) [inline fish-mode.el]: I wondered about that too.  I haven't noticed any problems, but I looked at `syntax-ppss`, and it is a rather large function, so I'll capture it in a variable like you suggested.

File: fish-mode.el (hunk 1 above)
Comment (alphapapa) [inline fish-mode.el]: Thanks, silly mistake on my part.

**Comment by alphapapa** [comment]
Latest commit fixes the problems you caught.  Thanks.

**Comment by wwwjfy** [comment]
LGTM. Thanks!

**Pull request merged**